src/android/MultiRegionService.java

```diff
@@ -0,0 +1,228 @@
+    }
+    protected synchronized void buildGoogleApiClient() {
+        mGoogleApiClient = new GoogleApiClient.Builder(this)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .addApi(LocationServices.API)
+                .build();
+    }
+    public void populateGeofenceList(String identifier, float lat, float lon, int radius){
+            mGeofenceList.add(new Geofence.Builder()
+                    // Set the request ID of the geofence. This is a string to identify this
+                    // geofence.
+                    .setRequestId(identifier)
+                            // Set the circular region of this geofence.
+                    .setCircularRegion(lat, lon, radius)
+                            // Set the expiration duration of the geofence. This geofence gets automatically
+                            // removed after this period of time.
+                    .setExpirationDuration(GEOFENCE_EXPIRATION_IN_MILLISECONDS)
+                            // Set the transition types of interest. Alerts are only generated for these
+                            // transition. We track entry and exit transitions in this sample.
+                    .setTransitionTypes(Geofence.GEOFENCE_TRANSITION_ENTER |
+                            Geofence.GEOFENCE_TRANSITION_EXIT)
+                            // Create the geofence.
+                    .build());
+
+    }
+
+    private void populateGeofenceRegion(String coord){
+        String identifer = "", lat = "", lon = "", radius = "";
+        String[] countRegion = coord.split(";");
+        String[] coordValue, regionValue;
+        int jk = 0;
+
+        for (int i = 0; i < countRegion.length; i++){
+            coordValue = countRegion[i].split(",");
+            for (int j = 0; j < coordValue.length; j++){
+                regionValue = coordValue[j].split(":");
+                for (int k = 0; k < regionValue.length; k++){
+                    if (k == 1) {
+                        switch (jk) {
+                            case 0: identifer = regionValue[k]; jk++;   break;
+                            case 1: lat = regionValue[k];       jk++;   break;
+                            case 2: lon = regionValue[k];       jk++;   break;
+                            case 3: radius = regionValue[k];    jk = 0; break;
+                        }
+                    }
+                }
+            }
+            populateGeofenceList(identifer, Float.parseFloat(lat), Float.parseFloat(lon), Integer.parseInt(radius));
+        }
+    }
+
+    public void addGeofences() {
+        if (!mGoogleApiClient.isConnected()) {
+            Log.w(TAG, "isConnected "+getString(R.string.not_connected));
+            return;
+        }
+
+        try {
+            LocationServices.GeofencingApi.addGeofences(
+                    mGoogleApiClient,
+                    // The GeofenceRequest object.
+                    getGeofencingRequest(),
```

Comment:
this looks like an endless loop, getGeofencingRequest calls addGeofences, which calls getGeofencingRequest...